Sources/TSCBasic/Process.swift

```diff
@@ -677,24 +688,40 @@ public final class Process {
     /// Blocks the calling process until the subprocess finishes execution.
     @discardableResult
     public func waitUntilExit() throws -> ProcessResult {
+        let group = DispatchGroup()
+        group.enter()
+        var processResult : Result<ProcessResult, Swift.Error>?
+        self.waitUntilExit() { result in
+            processResult = result
+            group.leave()
+        }
+        group.wait()
+        switch processResult! {
+        case .failure(let error):
+            throw error
+        case .success(let result):
+            return result
+        }
+    }
+
+    /// Executes the process I/O state machine, calling completion block when finished.
+    private func waitUntilExit(_ completion: @escaping (Result<ProcessResult, Swift.Error>) -> Void) {
```

Comment:
should we allow the caller to specify a queue instead of hard coding one? the caller may be some preferences

Reply:
Possibly. Before we were using a .global queue to wake up the dispatch group. Using a shared serial queue seems better. But a shared specified queue might be fine too.

Reply:
Added optional queue parameter in d67a7f0.